python/equistore-operations/equistore/operations/reduce_over_samples.py

```diff
@@ -258,22 +265,17 @@ def _reduce_over_samples_block(
                 if reduction == "var":
                     for i, s in enumerate(new_gradient_samples):
                         gradient_values_result[i] = (
-                            gradient_values_result[i] * values_mean[s[0]]
+                            gradient_values_result[i] * values_mean[int(s[0])]
                         )
                     gradient_values_result = 2 * (
                         values_grad_result - gradient_values_result
                     )
                 else:  # std
                     for i, s in enumerate(new_gradient_samples):
-                        # only numpy raise a warning for division by zero
-                        # so the statement catch that
-                        # for torch there is nothing to catch
-                        # both numpy and torch give inf for the division by zero
-                        with np.errstate(divide="ignore", invalid="ignore"):
```

Comment:
could we keep it by defining a function that does nothing for torch? Something like

```py
if torch.jit.is_scripting():
    def np_errstate(divide: str, invalid: str) -> None:
        return None
else:
    np_errstate = np.errstate
```

Reply:
The issue is that we don't want to import torch in that file, right?

Reply:
Plus, honestly, if the std is undefined, it's probably right that the user receives a warning instead of just a std of zero

Reply:
it is already exposed I think: `from ._classes import torch_jit_is_scripting`

Reply:
yes: https://github.com/lab-cosmo/equistore/blob/f09ec47369f71e4d71d02e372bd809da9ec5293f/python/equistore-operations/equistore/operations/equal_metadata.py#L6

Reply:
True. I've made a first attempt but the compiler is saying that, within a `with x():` expression, `x()` must return an object (see CI output)

Reply:
Found a solution